regulations/generator/html_builder.py

```diff
@@ -125,6 +95,59 @@ def process_node(self, node):
             node['label_id'],
             RegulationsConfig.node_type_tpls[node['node_type'].lower()])
 
+        for c in node['children']:
+            self.process_node(c)
+
+
+class CFRHTMLBuilder(HTMLBuilder):
+    header_regex = re.compile(r'^(§&nbsp;)(\s*\d+\.\d+)(.*)$')
+    section_number_regex = re.compile(r'(§+)\s+')
+
+    @classmethod
+    def section_space(cls, text):
+        """ After a section sign, insert a non-breaking space. """
+        return cls.section_number_regex.sub(r'\1&nbsp;', text)
+
+    def get_title(self):
+        titles = {
+            'part': self.tree['label'][0],
+            'reg_name': ''
+        }
+        reg_title = self.parse_doc_title(self.tree['title'])
+        if reg_title:
+            titles['reg_name'] = reg_title
+        return titles
+
+    def parse_doc_title(self, reg_title):
+        match = re.search(r"[(].+[)]$", reg_title)
```

Comment:
Nit-picking: why use character classes for parentheses here? Could we describe the same rule with the slightly simpler and arguably more idiomatic `\(.+\)$`? Also, I get that this just getting moved around and not new.